analyses/mutational-signatures/run_mutational_signatures.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+# J. Taroni for ALSF CCDL 
+# 2020
+
+set -e
+set -o pipefail
+
+# Set the working directory to the directory of this file
+cd "$(dirname "${BASH_SOURCE[0]}")"
+
+# In CI we'll run an abbreviated version of the de novo signatures extraction
+ABBREVIATED_MUTSIGS=${OPENPBTA_QUICK_MUTSIGS:-0}
+
+data_dir=../../data
+
+# Run the mutational signatures analysis using existing signatures
+Rscript -e "rmarkdown::render('01-mutational_signatures.Rmd', clean = TRUE)"
+
+# if abbreviated mutsigs is "true", we'll run only one number of signatures and
+# for an unacceptably low number of iterations to do any analysis with
+if [ "$ABBREVIATED_MUTSIGS" -gt "0" ]
+then
+  FLOOR=10
+  CEILING=10
+  NUM_ITER=10
+else
+  FLOOR=5
+  CEILING=15
+  NUM_ITER=1000
+fi
+
+# De novo signatures extraction
+Rscript --vanilla \
+  scripts/de_novo_signature_fitting.R \
+  --maf_file ${data_dir}/pbta-snv-consensus-mutation.maf.tsv.gz \
+  --nsignatures_floor $FLOOR \
+  --nsignatures_ceiling $CEILING \
+  --num_iterations $NUM_ITER \
+  --seed 42 \
+  --output_file results/denovo_sigfit_signatures.RDS
+
```

Comment:
That sounds good!
10,000 iterations leads to saving more digits within the object. In order to safe space, one could reduce the resolution in the code, as it is anyway a bit over the top.

Reply:
If I recall correctly, I thought the minimum number of recommended iterations was 2000 in the package documentation. Do you think something like 4000 would be appropriate? Is there an easy way to identify problems associated with too few iterations?

Reply:
I think 4000 should be fine, I just did 10000 to be on the safe side and it did not matter to need more resources. Reproducibility of the signatures would be a way to check.

Reply:
> Reproducibility of the signatures would be a way to check.

So run with a constant number of signatures and number of iterations but vary the seed and then calculate cosine similarity between runs, is that reasonable?

Reply:
Yes, it would be. However, it would be more standard to to measure reproducibility through average silhouette width rather than cosine similarity. The latter is usually used for comparing individual signatures. 
This paper details the methods for using average silhouette width: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146

Reply:
👍  great, thanks!